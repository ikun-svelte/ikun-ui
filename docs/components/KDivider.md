---
title: KDivider
lang: en-US
---

# KDivider

The dividing line that separates the content.

## Install

::: code-group

```bash [pnpm]
pnpm add @ikun-ui/divider
```

```bash [yarn]
yarn add @ikun-ui/divider
```

```bash [npm]
npm install @ikun-ui/divider
```

:::

## Basic usage

Divide the text of different paragraphs.

<demo src="divider/basic.svelte" github="Divider"></demo>

## Custom content

You can customize the content on the divider line.

<demo src="divider/custom-content.svelte" github="Divider"></demo>

## Line Style

You can set the style of divider.

<demo src="divider/line-style.svelte" github="Divider"></demo>

## Vertical divider

<demo src="divider/vertical-divider.svelte" github="Divider"></demo>

## Divider Props

| Name            | Type                                                                        | Default            | Description                                                 |
| --------------- | --------------------------------------------------------------------------- | ------------------ | ----------------------------------------------------------- |
| direction       | `horizontal` / `vertical`                                                   | `horizontal`       | Set divider's direction.                                    |
| borderColor     | `string`                                                                    | `'b-ikun-bd-base'` | divider's line color.                                       |
| borderStyle     | `dotted \| dashed \| solid \| double \| groove \| ridge \| inset \| outset` | `solid`            | divider's line style.                                       |
| contentPosition | `left` / `center` / `right`                                                 | `center`           | the position of the customized content on the divider line. |
| cls             | `string`                                                                    | `-`                | Additional class for component                              |
| attrs           | `Record<string, string>`                                                    | `{}`               | Additional attributes                                       |

## Divider Slots

| Name    | Description                             |
| ------- | --------------------------------------- |
| default | customized content on the divider line. |
