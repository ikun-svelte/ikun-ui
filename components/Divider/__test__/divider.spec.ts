import { tick } from 'svelte';
import { afterEach, expect, test, describe, beforeEach } from 'vitest';
import KDivider from '../src';
import KDividerContent from './fixture/divider.content.slot.test.svelte';

let host;

const initHost = () => {
	host = globalThis.document.createElement('div');
	host.setAttribute('id', 'host');
	globalThis.document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
});
afterEach(() => {
	host.remove();
});

describe('Test: KDivider', () => {
	test('props: cls', async () => {
		const instance = new KDivider({
			target: host,
			props: {
				cls: 'k-divider--test'
			}
		});
		expect(instance).toBeTruthy();
		expect(host!.innerHTML.includes('k-divider--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: direction', async () => {
		const instance = new KDivider({
			target: host,
			props: {
				direction: 'horizontal'
			}
		});
		expect(instance).toBeTruthy();
		expect(host!.innerHTML.includes('k-divider--horizontal')).toBeTruthy();
		instance.$set({ direction: 'vertical' });
		await tick();
		expect(host!.innerHTML.includes('k-divider--vertical')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: borderColor', async () => {
		const instance = new KDivider({
			target: host,
			props: {
				borderColor: 'b-red'
			}
		});
		expect(instance).toBeTruthy();
		expect(host!.innerHTML.includes('b-red')).toBeTruthy();
		instance.$set({ borderColor: 'b-blue' });
		await tick();
		expect(host!.innerHTML.includes('b-blue')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: borderStyle', async () => {
		const instance = new KDivider({
			target: host,
			props: {
				direction: 'horizontal',
				borderStyle: 'solid'
			}
		});
		expect(instance).toBeTruthy();
		expect(host!.innerHTML.includes('k-divider--horizontal__solid')).toBeTruthy();
		instance.$set({ borderStyle: 'dotted' });
		await tick();
		expect(host!.innerHTML.includes('k-divider--horizontal__dotted')).toBeTruthy();
		instance.$set({ direction: 'vertical', borderStyle: 'dashed' });
		await tick();
		expect(host!.innerHTML.includes('k-divider--vertical__dashed')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('slots: content', async () => {
		//@ts-ignore
		const instance = new KDividerContent({
			target: host
		});
		expect(instance).toBeTruthy();
		const dividers = host.querySelectorAll('.k-divider');
		expect(dividers[0].getAttribute('class')?.includes('k-divider--content-left')).toBeTruthy();
		expect(dividers[0].textContent).toBe('divider left text');
		expect(dividers[1].getAttribute('class')?.includes('k-divider--content-center')).toBeTruthy();
		expect(dividers[1].textContent).toBe('divider center text');
		expect(dividers[2].getAttribute('class')?.includes('k-divider--content-right')).toBeTruthy();
		expect(dividers[2].textContent).toBe('divider right text');
		expect(host.innerHTML).matchSnapshot();
	});
});
