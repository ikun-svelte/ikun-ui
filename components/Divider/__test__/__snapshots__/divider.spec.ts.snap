// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KDivider > props: borderColor 1`] = `"<div class=\\"k-divider k-divider--horizontal k-divider--horizontal__solid k-divider--content-center b-blue\\" data-divider=\\"k-divider\\"></div>"`;

exports[`Test: KDivider > props: borderStyle 1`] = `"<div class=\\"k-divider k-divider--vertical k-divider--vertical__dashed k-divider--content-center b-ikun-bd-base\\" data-divider=\\"k-divider\\"></div>"`;

exports[`Test: KDivider > props: cls 1`] = `"<div class=\\"k-divider k-divider--horizontal k-divider--horizontal__solid k-divider--content-center b-ikun-bd-base k-divider--test\\" data-divider=\\"k-divider\\"></div>"`;

exports[`Test: KDivider > props: direction 1`] = `"<div class=\\"k-divider k-divider--vertical k-divider--vertical__solid k-divider--content-center b-ikun-bd-base\\" data-divider=\\"k-divider\\"></div>"`;

exports[`Test: KDivider > slots: content 1`] = `"<div class=\\"k-divider k-divider--horizontal k-divider--horizontal__solid k-divider--content-left b-ikun-bd-base\\" data-divider=\\"k-divider\\"><span class=\\"k-divider--content\\">divider left text</span></div> <div class=\\"k-divider k-divider--horizontal k-divider--horizontal__solid k-divider--content-center b-ikun-bd-base\\" data-divider=\\"k-divider\\"><span class=\\"k-divider--content\\">divider center text</span></div> <div class=\\"k-divider k-divider--horizontal k-divider--horizontal__solid k-divider--content-right b-ikun-bd-base\\" data-divider=\\"k-divider\\"><span class=\\"k-divider--content\\">divider right text</span></div>"`;
